### DISCUSSION TITLE
use-a-hook-to-hide-modules-for-non-admin-roles-in-directus/

### DISCUSSION THREAD

**Discussion by martijnboland** [martijnboland/blog#60] [Blog comments]
# use-a-hook-to-hide-modules-for-non-admin-roles-in-directus/

https://blogs.taiga.nl/martijn/posts/use-a-hook-to-hide-modules-for-non-admin-roles-in-directus/

**Comment by disqus-comments-migrator** [comment]
<em>[Ben Amato](https://disqus.com/by/ben_amato/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 10 2023, 01:40</em>

---

<p>thanks Martijn!</p>

**Comment by disqus-comments-migrator** [comment]
<em>[Matt](https://disqus.com/by/disqus_OOVqKoNNKP/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at maart 25 2023, 06:35</em>

---

<p>Thanks! Can you clarify if this is a visual/cosmetic fix only, or if it is secure (any way for people to "unhide" modules by opening up chrome inspector and changing things, or any other way?) Also, as a new user I would love to learn more about the specific way to  check for specific roles, or filtering the existing module_bar items as you mentioned!</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at maart 29 2023, 10:07, in reply to [Matt](https://disqus.com/by/disqus_OOVqKoNNKP/)'s [comment](https://github.com/martijnboland/blog/discussions/60#discussioncomment-8953108)</em>

---

<p>Hi Matt, this is a secure fix at the server-side, so no obscuring or whatever. I the hook, you have full access to the current user and their roles, so you could customize it completely on a per-role basis.</p>

**Comment by disqus-comments-migrator** [comment]
<em>[Stefano Bolzonetti](https://disqus.com/by/stefanobolzonetti/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at november 09 2023, 11:59</em>

---

<p>Does it also work with the latest versions? because I tried it and it doesn't work.</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at november 09 2023, 12:17, in reply to [Stefano Bolzonetti](https://disqus.com/by/stefanobolzonetti/)'s [comment](https://github.com/martijnboland/blog/discussions/60#discussioncomment-8953113)</em>

---

<p>I haven't tried it in the latest versions, so it might not work anymore. I'll check if I an find some time.</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at december 15 2023, 01:13, in reply to [Stefano Bolzonetti](https://disqus.com/by/stefanobolzonetti/)'s [comment](https://github.com/martijnboland/blog/discussions/60#discussioncomment-8953113)</em>

---

<p>Just tried with Directus 1.8.2 and it still works (on my machine though)</p>
